network.cpp

```diff
@@ -64,25 +64,31 @@ bool operator!=(const network& lhs, const network& rhs)
 
 network change_all_weights(network n, double new_weight)
 {
-    for(auto& layer : n.get_net_weights())
-        for(auto& node : layer)
-            for(auto& weight : node)
-            {
-                weight.change_weight(new_weight);
+  auto new_weights = n.get_net_weights();
+
+  for(auto& layer : new_weights)
+      for(auto& node : layer)
+          for(auto& weight : node){
+            weight.change_weight(new_weight);
```

Comment:
maybe we can even rename this member function of weight `change()`, it even sounds more english-like. Optional.

Reply:
I like having `change_weight()` because there's also `change_activation()`. Honestly, I think what would make most sense is renaming the whole` weight` class to` connection`. What do you think about it?

Reply:
👍